src/hal/src/external_memory/mod.rs

```diff
@@ -0,0 +1,498 @@
+//! Structures related to the import external memory functionality
+
+mod errors;
+pub use errors::*;
+
+pub use external_memory::*;
+
+#[cfg(any(target_os = "linux", target_os = "android"))]
+pub use drm_fourcc::*;
+
+/// Buffer or image
+#[derive(Debug)]
+pub enum Resource<B: crate::Backend> {
+    /// Buffer
+    Buffer(B::Buffer),
+    /// Image
+    Image(B::Image),
+}
+
+/// External memory properties
+#[derive(Debug, PartialEq)]
+pub struct ExternalMemoryProperties {
+    exportable: bool,
+    importable: bool,
+    exportable_from_imported: bool,
+}
+impl ExternalMemoryProperties {
+    /// Constructor
+    pub fn new(exportable: bool, importable: bool, exportable_from_imported: bool) -> Self {
+        Self {
+            exportable,
+            importable,
+            exportable_from_imported,
+        }
+    }
+    /// Is the queried configuration exportable
+    pub fn is_exportable(&self) -> bool {
```

Comment:
I don't think we should go out of our way to encapsulate the immutability of these fields. Just follow the existing pattern. We have memory requirements, and they are returned just by value, with all the fields public.

Reply:
Since the structure has become a bitflag, that functions has been removed, they are no more needed.